lib/ui/platform_dispatcher.dart

```diff
@@ -308,6 +308,93 @@ class PlatformDispatcher {
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
+  /// A callback invoked immediately after the focus is transitioned across [FlutterView]s.
+  ///
+  /// When the platform moves the focus from one [FlutterView] to another, this
+  /// callback is invoked indicating the new view that has focus and the direction
+  /// in which focus was received. For example, if focus is moved to the [FlutterView]
+  /// with ID 2 in the forward direction (could be the result of pressing tab)
+  /// the callback receives the following [ViewFocusEvent]:
+  ///
+  /// ```dart
+  /// ViewFocusEvent(
+  ///   viewId: 2,
+  ///   state: ViewFocusState.focused,
+  ///   direction: ViewFocusDirection.forward,
+  /// )
+  /// ```
+  ///
+  /// Typically, receivers of this event respond by moving the focus to the first
+  /// focusable element inside the [FlutterView] with ID 2. If a view receives
+  /// focus in the backwards direction (could be the result of pressing shift + tab),
+  /// typically the last focusable element inside that view is focused.
+  ///
+  /// The platform can also move input focus to a non-[FlutterView] element.
+  /// In that case, this callback is invoked with [ViewFocusState.unfocus] and
+  /// [ViewFocusDirection.undefined].
+  ///
+  /// ```dart
+  /// ViewFocusEvent(
+  ///   viewId: 2,
+  ///   state: ViewFocusState.unfocused,
+  ///   direction: ViewFocusDirection.undefined,
+  /// )
+  /// ```
+  ///
+  /// Receivers typically respond to this event by removing all focus indications
+  /// from the app.
+  ///
+  /// Apps can also programmatically request to move the focus to a desired
+  /// [FlutterView] by calling [requestViewFocusChange].
+  ///
+  /// The callback is invoked in the same zone in which the callback was set.
+  ///
+  /// See also:
+  ///
+  ///   * [requestViewFocusChange] to programmatically instruct the platform to move focus to a different [FlutterView].
+  ///   * [ViewFocusState] for a list of allowed focus transitions.
+  ///   * [ViewFocusDirection] for a list of allowed focus directions.
+  ///   * [ViewFocusEvent], which is the event object provided to the callback.
+  ViewFocusChangeCallback? get onViewFocusChange => _onViewFocusChange;
+  ViewFocusChangeCallback? _onViewFocusChange;
+  // ignore: unused_field, field will be used when platforms other than web use these focus APIs.
+  Zone _onViewFocusChangeZone = Zone.root;
+  set onViewFocusChange(ViewFocusChangeCallback? callback) {
+    _onViewFocusChange = callback;
+    _onViewFocusChangeZone = Zone.current;
+  }
+
+  /// Requests a focus change of the flutter view with ID [viewId].
```

Comment:
nit: s/flutter/Flutter/

However, I'd probably not even mention "Flutter". Just "view" is sufficient.